cs/ObjectPool.cs

```diff
@@ -157,6 +160,8 @@ public void UpdateFactory(ObjectFactory<TSource, TObject> factory)
 
         internal void ReturnObject(PooledObject<TSource, TObject> pooledObject)
         {
+            Contract.Ensures(pooledObject != null);
```

Comment:
remind me whether this will throw a runtime exception? I think the last time I tried it didn't do anything...